tests/components/min_max/test_sensor.py

```diff
@@ -14,329 +13,334 @@
-        assert self.values[1] == state.attributes.get("max_value")
-
-        self.hass.states.set(entity_ids[1], STATE_UNAVAILABLE)
-        self.hass.block_till_done()
-
-        state = self.hass.states.get("sensor.test_max")
-        assert STATE_UNKNOWN == state.state
-        assert state.attributes.get("min_entity_id") is None
-        assert state.attributes.get("min_value") is None
-        assert state.attributes.get("max_entity_id") is None
-        assert state.attributes.get("max_value") is None
-
-    def test_different_unit_of_measurement(self):
-        """Test for different unit of measurement."""
-        config = {
-            "sensor": {
-                "platform": "min_max",
-                "name": "test",
-                "type": "mean",
-                "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
-            }
+    hass.states.async_set(entity_ids[2], STATE_UNKNOWN)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.test_max")
+    assert STATE_UNKNOWN != state.state
+    assert entity_ids[1] == state.attributes.get("min_entity_id")
+    assert VALUES[1] == state.attributes.get("min_value")
+    assert entity_ids[1] == state.attributes.get("max_entity_id")
+    assert VALUES[1] == state.attributes.get("max_value")
+
+    hass.states.async_set(entity_ids[1], STATE_UNAVAILABLE)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.test_max")
+    assert STATE_UNKNOWN == state.state
+    assert state.attributes.get("min_entity_id") is None
+    assert state.attributes.get("min_value") is None
+    assert state.attributes.get("max_entity_id") is None
+    assert state.attributes.get("max_value") is None
+
+
+async def test_different_unit_of_measurement(hass):
+    """Test for different unit of measurement."""
+    config = {
+        "sensor": {
+            "platform": "min_max",
+            "name": "test",
+            "type": "mean",
+            "entity_ids": ["sensor.test_1", "sensor.test_2", "sensor.test_3"],
         }
+    }
 
-        assert setup_component(self.hass, "sensor", config)
+    assert await async_setup_component(hass, "sensor", config)
+    await hass.async_block_till_done()
 
-        entity_ids = config["sensor"]["entity_ids"]
+    entity_ids = config["sensor"]["entity_ids"]
 
-        self.hass.states.set(
-            entity_ids[0], self.values[0], {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
-        )
-        self.hass.block_till_done()
+    hass.states.async_set(
+        entity_ids[0], VALUES[0], {ATTR_UNIT_OF_MEASUREMENT: TEMP_CELSIUS}
+    )
+    await hass.async_block_till_done()
 
-        state = self.hass.states.get("sensor.test")
+    state = hass.states.get("sensor.test")
 
-        assert str(float(self.values[0])) == state.state
-        assert state.attributes.get("unit_of_measurement") == TEMP_CELSIUS
+    assert str(float(VALUES[0])) == state.state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)